ndc-lambda-sdk/src/execution.ts

```diff
@@ -167,6 +170,38 @@ async function invokeFunction(func: Function, preparedArgs: unknown[], functionN
   }
 }
 
+export type ErrorDetails = {
+  message: string
+  stack: string
+}
+
+export function getErrorDetails(error: Error): ErrorDetails {
+  return {
+    message: error.message,
+    stack: buildCausalStackTrace(error),
+  }
+}
+
+function buildCausalStackTrace(error: Error): string {
+  let currErr: Error | undefined = error;
+  let stackTrace = "";
+
+  while (currErr) {
+    if (currErr.stack) {
+      stackTrace += `${currErr.stack}${EOL}`;
+    } else {
+      stackTrace += `${currErr.toString()}${EOL}`;
+    }
+    if (currErr.cause instanceof Error) {
+      stackTrace += `caused by `;
+      currErr = currErr.cause;
+    } else {
+      currErr = undefined;
+    }
+  }
+  return stackTrace;
+}
```

Comment:
Could this ever get into an infinite loop?

Reply:
Fixed.